File: src/dotnet/Ansight.Core/Pairing/Models/PairingConfigDocument.cs
namespace Ansight.Pairing.Models;

/// <summary>
/// Self-contained enrollment invite used by QR scanning.
/// </summary>
public sealed class PairingConfigDocument
{
    /// <summary>
    /// Current schema identifier for pairing config document payloads.
    /// </summary>
    public const string SchemaName = "ansight.enrollment-invite-document.v2";

    /// <summary>
    /// Schema identifier for this config document.
    /// </summary>
    public string Schema { get; set; } = SchemaName;

    /// <summary>
    /// Enrollment invite carried by the document.
    /// </summary>
    [System.Text.Json.Serialization.JsonPropertyName("invite")]
    public required PairingConfig Config { get; set; }

    /// <summary>
    /// Discovery metadata that helps the client reach the host.
    /// </summary>
    public PairingDiscoveryHint? Discovery { get; set; }
}

File: src/ios/Sources/AnsightCore/PairingConfigDocument.swift
import Foundation

public struct PairingConfigDocument: Sendable, Codable, Equatable {
    public static let schemaName = "ansight.enrollment-invite-document.v2"

    public var schema: String
    public var config: PairingConfig
    public var discovery: PairingDiscoveryHint?

    private enum CodingKeys: String, CodingKey {
        case schema
        case config = "invite"
        case discovery
    }

    public init(
        schema: String = PairingConfigDocument.schemaName,
        config: PairingConfig,
        discovery: PairingDiscoveryHint? = nil
    ) {
        self.schema = schema
        self.config = config
        self.discovery = discovery
    }
}
